core/governance/market.go

```diff
@@ -271,10 +270,23 @@ func validateFuture(future *types.FutureProduct, decimals uint64, assets Assets,
 	if err != nil {
 		return types.ProposalErrorInvalidFutureProduct, err
 	}
-	if err := ospec.EnsureBoundableProperty(future.DataSourceSpecBinding.SettlementDataProperty, datapb.PropertyKey_TYPE_INTEGER); err != nil {
-		return types.ProposalErrorInvalidFutureProduct, fmt.Errorf("invalid oracle spec binding for settlement data: %w", err)
+
+	switch future.DataSourceSpecBinding.SettlementDataProperty {
+	case datapb.PropertyKey_TYPE_DECIMAL.String():
+		err := ospec.EnsureBoundableProperty(future.DataSourceSpecBinding.SettlementDataProperty, datapb.PropertyKey_TYPE_DECIMAL)
+		if err != nil {
+			return types.ProposalErrorInvalidFutureProduct, fmt.Errorf("invalid oracle spec binding for settlement data: %w", err)
+		}
+	case datapb.PropertyKey_TYPE_INTEGER.String():
+		fallthrough
```

Comment:
Why specify the case if it's a fallthrough to the default?

Reply:
I left these to aim more clarity/readability when reading the code, but I guess no one needed that (except me); Addressed this and the below 2 as well: https://github.com/vegaprotocol/vega/pull/6935/commits/14c69f3c1ab4f4a6fdbaffe0ee6d971d6658e4f8#diff-1527092608a66a12e24c11a95fc2ca1e9d2b73063b35c78ad06651b45b37bef8R281
Thanks!